
Analyses: We provide a variety of standard and custom sequencing data analyses

```{r, echo = FALSE, message = FALSE}
dat <- tibble(
  analysis = c(
    
  "Transcriptomics",
  "Single-cell analytics",
  "Spatial genomics",
  "Spatial genomics",	
  "Genomics",
  "Genomics",	
  "Epigenomics", 	
  "Metagenomics", 	
  "Custom analyses", 	
  "Public Datasets",	
  "Functional Analyses", 	
  "Functional Analyses", 	
  "Consultations",
  "GEO/SRA uploads")
  ,
  description = c(
    "RNA-Seq, UMI-based RNA-Seq, High Throughput RNA-Seq (”Drug-Seq”), smallRNA-Seq (miRNA-Seq)",  
    "Gene expression, Immune profiling (VDJ, VDJ+Gene expression), Epigenome profiling (ATAC, Multiome ATAC + Gene expression)",
    "SNP-based demultiplexing of multiplexed data",
    "Visium Spatial transcriptomics (10x Genomics)",
    "GeoMx Digital Spatial Profiling (nanoString)",
    "Exome, Whole Genome sequencing data analyses, SNPs, indels, CNVs",
    "ChIP-Seq/eCLIP-Seq/CUT&RUN/RIP-Seq,  ATAC-Seq",
    "16S rRNA-Seq",
    "Sequencing data analyses for specific lab projects that use standard tools, software packages and/or custom code/script developed by CCBB.",
    "Analyses on public dataset available at NCBI’s GEO database repository",
    "Advaita’s iPathwayGuide for a full comprehensive functional analysis including pathways/GO terms/mRNAs/upstream regulators/diseases ADVAITA service request",
    "Gene Set Enrichment Analysis (GSEA)",
    "We provide consultations on experimental design, overview of analyses and costs involved. We can take on both short-term (typically less than a month and about 10 hours of CCBB effort) and long-term collaborative projects involving bioinformatic support.",
    "We can submit your sequencing data, related sample metadata and the results of analyses to NCBI repositories like Gene Expression Omnibus (GEO) and Sequence Read Archive (SRA) for a small service fee."
  )
)

knitr::kable(dat)
```



